online-shop/src/app/store/reducers/product.reducers.ts

```diff
@@ -0,0 +1,49 @@
+import {initialProductState, IProductState} from "../state/product.state";
+import {EProductActions, ProductActions} from "../actions/product.actions";
+
+export const productReducers = (
+  state = initialProductState,
+  action: ProductActions
+): IProductState => {
+  switch (action.type) {
+    case EProductActions.GetProductsSuccess: {
+      return {
+        ...state,
+        products: action.payload
+      };
+    }
+
+    case EProductActions.GetProductSuccess: {
+      return {
+        ...state,
+        selectedProduct: action.payload
+      };
+    }
+
+    case EProductActions.DeleteProductSuccess: {
+      return {
+        ...state,
+        products: state.products.filter(
```

Comment:
This is a service-job. Your component dispatches an action, goes to effects, here we call the service where we do the backend call, and after you can simply dispatch a new action from effects. This, ofc, is a very simple example and you can do it using lists and simple operations. But take this into consideration for your future work